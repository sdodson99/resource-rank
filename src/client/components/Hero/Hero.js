import React, { useState } from 'react';
import styles from './Hero.module.css';
import { useRouter } from 'next/router';

const Hero = () => {
  const [searchTopicsQuery, setSearchTopicsQuery] = useState('');
  const router = useRouter();

  const onSearchTopicsQueryChange = (e) => {
    const value = e.target.value;

    setSearchTopicsQuery(value);
  };

  const onSearchTopics = () => {
    router.push(`/topics?q=${searchTopicsQuery}`);
  };

  return (
    <div className={styles.Hero} data-testid="Hero">
      <div className={`content-container ${styles.Content}`}>
        <div className="text-7xl">What are you planning to learn?</div>
        <div className={styles.Search}>
          <input
            type="text"
            className="flex-grow block form-control-lg"
            placeholder="Search topics..."
            value={searchTopicsQuery}
            onChange={onSearchTopicsQueryChange}
          />
          <button
            className="mt-5 sm:mt-0 sm:ml-8 btn btn-primary"
            onClick={onSearchTopics}
          >
            Search
          </button>
        </div>
        <div className="mt-24">
          <div className="text-2xl">
            <strong>Resource Rank</strong> helps you find the{' '}
            <strong>best</strong> resources for your learning journey.
          </div>
          <div className="mt-8 flex justify-center">
            <img
              src="/img/logo.svg"
              alt="Resource Rank Logo"
              height="100"
              width="100"
            />
          </div>
        </div>
      </div>
      <div className={styles.ColorOverlay}></div>
    </div>
  );
};

Hero.propTypes = {};

Hero.defaultProps = {};

export default Hero;
