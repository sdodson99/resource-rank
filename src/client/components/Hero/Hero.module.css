.Hero {
  @apply py-16 text-center bg-gray-100 md:text-left;
}

.PromptSection {
  flex: 2;
}

.IllustrationSection {
  flex: 1;

  @apply mt-16 md:mt-0 md:ml-10;
}

.Prompt {
  @apply text-7xl;
}

.SubPrompt {
  @apply mt-12 text-2xl;
}

.Emphasis {
  color: var(--color-primary);
}

.Illustration {
  max-height: 300px;
}
