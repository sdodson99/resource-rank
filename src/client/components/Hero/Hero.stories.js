import React from 'react';
import Hero from './Hero';

export default {
  component: Hero,
  title: 'Components/Hero',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <Hero {...args} />;

export const Default = Template.bind({});
Default.args = {};
