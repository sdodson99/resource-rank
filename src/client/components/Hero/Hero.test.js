import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Hero from './Hero';
import { useRouter } from 'next/router';

jest.mock('next/router');

describe('<Hero />', () => {
  let mockRouterPush;

  beforeEach(() => {
    mockRouterPush = jest.fn();

    useRouter.mockReturnValue({
      push: mockRouterPush,
    });
  });

  afterEach(() => {
    useRouter.mockReset();
  });

  test('it should mount', () => {
    render(<Hero />);

    const hero = screen.getByTestId('Hero');

    expect(hero).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<Hero />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should navigate to topic search when search button clicked', () => {
    const search = 'topic123';
    render(<Hero />);
    const searchInput = screen.getByPlaceholderText('Search topics...');
    const searchButton = screen.getByText('Search');

    fireEvent.change(searchInput, {
      target: {
        value: search,
      },
    });
    searchButton.click();

    expect(mockRouterPush).toBeCalledWith(`/topics?q=${search}`);
  });
});
