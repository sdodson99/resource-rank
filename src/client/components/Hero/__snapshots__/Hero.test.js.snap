// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should render correctly 1`] = `
<div
  className="Hero"
  data-testid="Hero"
>
  <div
    className="content-container Content"
  >
    <div
      className="text-7xl"
    >
      What are you planning to learn?
    </div>
    <div
      className="Search"
    >
      <input
        className="flex-grow block form-control-lg"
        onChange={[Function]}
        placeholder="Search topics..."
        type="text"
        value=""
      />
      <button
        className="mt-5 sm:mt-0 sm:ml-8 btn btn-primary"
        onClick={[Function]}
      >
        Search
      </button>
    </div>
    <div
      className="mt-24"
    >
      <div
        className="text-2xl"
      >
        <strong>
          Resource Rank
        </strong>
         helps you find the
         
        <strong>
          best
        </strong>
         resources for your learning journey.
      </div>
      <div
        className="mt-8"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 100,
              "overflow": "hidden",
              "position": "relative",
              "width": 100,
            }
          }
        >
          <noscript>
            <img
              alt="Resource Rank Logo"
              decoding="async"
              src="/img/logo.svg"
              srcSet="/_next/image?url=%2Fimg%2Flogo.svg&w=128&q=75 1x, /_next/image?url=%2Fimg%2Flogo.svg&w=256&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
          <img
            alt="Resource Rank Logo"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ColorOverlay"
  />
</div>
`;
